airflow/www/views.py

```diff
@@ -2293,15 +2300,17 @@ def clear(self, *, session: Session = NEW_SESSION):
 
             # Lock the related dag runs to prevent from possible dead lock.
             # https://github.com/apache/airflow/pull/26658
-            dag_runs_query = session.query(DagRun.id).filter(DagRun.dag_id == dag_id).with_for_update()
+            dag_runs_query = session.scalars(
+                select(DagRun.id).where(DagRun.dag_id == dag_id).with_for_update()
+            )
             if start_date is None and end_date is None:
-                dag_runs_query = dag_runs_query.filter(DagRun.execution_date == start_date)
+                dag_runs_query = dag_runs_query.where(DagRun.execution_date == start_date)
             else:
                 if start_date is not None:
-                    dag_runs_query = dag_runs_query.filter(DagRun.execution_date >= start_date)
+                    dag_runs_query = dag_runs_query.where(DagRun.execution_date >= start_date)
```

Comment:
I'm not too familiar with sqlalchemy 2.0 but this line is throwing an error and is breaking the ability to clear task groups in the UI
`AttributeError: 'ScalarResult' object has no attribute 'where'`

Reply:
Fixed in #32801